src/ComplexNumber.java

```diff
@@ -0,0 +1,36 @@
+public final class ComplexNumber {
+    private final double re;
+    private final double im;
+
+    public ComplexNumber(double re, double im) {
+        this.re = re;
+        this.im = im;
+    }
+
+    public double getRe() {
+        return re;
+    }
+
+    public double getIm() {
+        return im;
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj) return true;
+        if (obj instanceof ComplexNumber) {
```

Comment:
where is the checking of the null?

Reply:
"instanceof" checked of the null, but if you want I may add checking obviously "if (obj != null && obj instanceof ComplexNumber)"

Reply:
no need to add checking. It is ok